0.13.3 IllegalArgumentException: argument type mismatch

jazzer version 0.13.3
args
```
--autofuzz=ru.vyarus.java.generics.resolver.GenericsResolver::resolve --ignore=c803d27b56af8836,79a165404bfc8505
```
target version:
```xml
<dependency>
          <groupId>ru.vyarus</groupId>
          <artifactId>generics-resolver</artifactId>
          <version>3.0.1</version>
<dependency>
```
```
INFO: Loaded 126 hooks from com.code_intelligence.jazzer.runtime.TraceCmpHooks
INFO: Loaded 4 hooks from com.code_intelligence.jazzer.runtime.TraceDivHooks
INFO: Loaded 2 hooks from com.code_intelligence.jazzer.runtime.TraceIndirHooks
INFO: Loaded 4 hooks from com.code_intelligence.jazzer.runtime.NativeLibHooks
INFO: Loaded 8 hooks from com.code_intelligence.jazzer.sanitizers.Deserialization
INFO: Loaded 5 hooks from com.code_intelligence.jazzer.sanitizers.ExpressionLanguageInjection
INFO: Loaded 70 hooks from com.code_intelligence.jazzer.sanitizers.LdapInjection
INFO: Loaded 46 hooks from com.code_intelligence.jazzer.sanitizers.NamingContextLookup
INFO: Loaded 1 hooks from com.code_intelligence.jazzer.sanitizers.OsCommandInjection
INFO: Loaded 76 hooks from com.code_intelligence.jazzer.sanitizers.ReflectiveCall
INFO: Loaded 8 hooks from com.code_intelligence.jazzer.sanitizers.RegexInjection
INFO: Loaded 16 hooks from com.code_intelligence.jazzer.sanitizers.RegexRoadblocks
INFO: Loaded 19 hooks from com.code_intelligence.jazzer.sanitizers.SqlInjection
INFO: Instrumented net.ricecode.similarity.LevenshteinDistanceStrategy (took 259 ms, size +33%)
INFO: Instrumented net.ricecode.similarity.StringSimilarityServiceImpl (took 47 ms, size +10%)
INFO: Instrumented net.ricecode.similarity.StringSimilarityService (took 3 ms, size +0%)
INFO: Instrumented net.ricecode.similarity.SimilarityStrategy (took 2 ms, size +0%)
INFO: Instrumented fr.xephi.authme.util.StringUtils (took 68 ms, size +55%)
INFO: Instrumented test.Fuzzer (took 67 ms, size +16%)
INFO: Instrumented ru.vyarus.java.generics.resolver.GenericsResolver (took 34 ms, size +8%)
INFO: Instrumented ru.vyarus.java.generics.resolver.context.GenericsContext (took 239 ms, size +8%)
INFO: Instrumented ru.vyarus.java.generics.resolver.context.AbstractGenericsContext (took 189 ms, size +9%)
INFO: libFuzzer ignores flags that start with '--'
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 2547083470
INFO: Loaded 1 modules   (512 inline 8-bit counters): 512 [0x7f87748145a0, 0x7f87748147a0), 
INFO: Loaded 1 PC tables (512 PCs): 512 [0x7f877480ec30,0x7f8774810c30), 
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: Instrumented ru.vyarus.java.generics.resolver.util.TypeUtils (took 84 ms, size +21%)
INFO: Instrumented ru.vyarus.java.generics.resolver.util.walk.TypesVisitor (took 8 ms, size +0%)
INFO: Instrumented ru.vyarus.java.generics.resolver.util.map.IgnoreGenericsMap (took 56 ms, size +22%)
INFO: Instrumented ru.vyarus.java.generics.resolver.util.TypeUtils$1 (took 12 ms, size +15%)
INFO: A corpus is not provided, starting from an empty corpus
INFO: Instrumented ru.vyarus.java.generics.resolver.error.UnknownGenericException (took 45 ms, size +24%)
INFO: Instrumented ru.vyarus.java.generics.resolver.error.GenericSourceException (took 30 ms, size +11%)
INFO: Instrumented ru.vyarus.java.generics.resolver.error.GenericsException (took 10 ms, size +17%)
INFO: Instrumented ru.vyarus.java.generics.resolver.context.MethodGenericsContext (took 84 ms, size +8%)
INFO: Instrumented ru.vyarus.java.generics.resolver.context.ConstructorGenericsContext (took 50 ms, size +9%)
INFO: New number of coverage counters: 1024
INFO: Instrumented ru.vyarus.java.generics.resolver.error.WrongGenericsContextException (took 57 ms, size +9%)
INFO: Instrumented ru.vyarus.java.generics.resolver.context.GenericsInfo$TypeWriter (took 3 ms, size +0%)
INFO: Instrumented ru.vyarus.java.generics.resolver.util.map.PrintableGenericsMap (took 14 ms, size +32%)
INFO: Instrumented ru.vyarus.java.generics.resolver.context.GenericsInfoFactory (took 38 ms, size +45%)
```
```
#2	INITED cov: 8 ft: 8 corp: 1/1b exec/s: 0 rss: 2260Mb
Unexpected exception encountered during autofuzz
com.code_intelligence.jazzer.autofuzz.AutofuzzError: 
Method: ru.vyarus.java.generics.resolver.GenericsResolver::resolve(java.lang.Class,java.lang.Class[])
this: null
Arguments: null, [Ljava.lang.Object;@1cf6d1be
Please file an issue at:
  https://github.com/CodeIntelligenceTesting/jazzer/issues/new/choose
	at com.code_intelligence.jazzer.autofuzz.Meta.autofuzz(Meta.java:277)
	at com.code_intelligence.jazzer.autofuzz.Meta.autofuzz(Meta.java:233)
	at com.code_intelligence.jazzer.autofuzz.FuzzTarget.fuzzerTestOneInput(FuzzTarget.java:261)
	at com.code_intelligence.jazzer.autofuzz.FuzzTarget.fuzzerTestOneInput(FuzzTarget.java:226)
	at com.code_intelligence.jazzer.driver.FuzzTargetRunner.runOne(FuzzTargetRunner.java:184)
	at com.code_intelligence.jazzer.runtime.FuzzTargetRunnerNatives.startLibFuzzer(Native Method)
	at com.code_intelligence.jazzer.driver.FuzzTargetRunner.startLibFuzzer(FuzzTargetRunner.java:409)
	at com.code_intelligence.jazzer.driver.FuzzTargetRunner.startLibFuzzer(FuzzTargetRunner.java:282)
	at com.code_intelligence.jazzer.driver.Driver.start(Driver.java:88)
	at com.code_intelligence.jazzer.Jazzer.start(Jazzer.java:90)
	at com.code_intelligence.jazzer.Jazzer.main(Jazzer.java:63)
	at test.Fuzzer.main(Unknown Source)
Caused by: java.lang.IllegalArgumentException: argument type mismatch
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:108)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at com.code_intelligence.jazzer.autofuzz.Meta.autofuzz(Meta.java:274)
	... 11 more
==6523== ERROR: libFuzzer: fuzz target exited
SUMMARY: libFuzzer: fuzz target exited
MS: 1 CMP- DE: "\001\000"-; base unit: adc83b19e793491b1c6ea0fd8b46cd9f32e592fc
0xa,0x1,0x0,
\012\001\000
artifact_prefix='./'; Test unit written to ./crash-4571ef1e60c5884b09f4fa9e1d366e6308ff8ea9
Base64: CgEA
```

--- a/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java
+++ b/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java
@@ -202,7 +202,7 @@ public class Meta {
         Arrays.stream(arguments).map(Meta::deepToString).collect(Collectors.joining(", ")));
   }
 
-  private static Class<?> getRawType(Type genericType) {
+  static Class<?> getRawType(Type genericType) {
     if (genericType instanceof Class<?>) {
       return (Class<?>) genericType;
     } else if (genericType instanceof ParameterizedType) {
@@ -213,8 +213,9 @@ public class Meta {
     } else if (genericType instanceof TypeVariable<?>) {
       throw new AutofuzzError("Did not expect genericType to be a TypeVariable: " + genericType);
     } else if (genericType instanceof GenericArrayType) {
-      // TODO: Improve this;
-      return Object[].class;
+      return Array
+          .newInstance(getRawType(((GenericArrayType) genericType).getGenericComponentType()), 0)
+          .getClass();
     } else {
       throw new AutofuzzError("Got unexpected class implementing Type: " + genericType);
     }

--- a/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java
+++ b/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java
@@ -183,14 +183,23 @@ public class Meta {
     return bytesToSpend / Math.max(sizeOfElement, 1);
   }
 
+  private static String deepToString(Object obj) {
+    if (obj == null) {
+      return "null";
+    }
+    if (obj.getClass().isArray()) {
+      return String.format("(%s[]) %s", obj.getClass().getComponentType().getName(),
+          Arrays.deepToString((Object[]) obj));
+    }
+    return obj.toString();
+  }
+
   private static String getDebugSummary(
       Executable executable, Object thisObject, Object[] arguments) {
     return String.format("%nMethod: %s::%s%s%nthis: %s%nArguments: %s",
         executable.getDeclaringClass().getName(), executable.getName(),
         Utils.getReadableDescriptor(executable), thisObject,
-        Arrays.stream(arguments)
-            .map(arg -> arg == null ? "null" : arg.toString())
-            .collect(Collectors.joining(", ")));
+        Arrays.stream(arguments).map(Meta::deepToString).collect(Collectors.joining(", ")));
   }
 
   private static Class<?> getRawType(Type genericType) {

--- a/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java
+++ b/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java
@@ -328,43 +328,51 @@ public class Meta {
     Class<?> type = getRawType(genericType);
     if (type == byte.class || type == Byte.class) {
       byte result = data.consumeByte();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(String.format("(byte) %s", result));
+      }
       return result;
     } else if (type == short.class || type == Short.class) {
       short result = data.consumeShort();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(String.format("(short) %s", result));
+      }
       return result;
     } else if (type == int.class || type == Integer.class) {
       int result = data.consumeInt();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(Integer.toString(result));
+      }
       return result;
     } else if (type == long.class || type == Long.class) {
       long result = data.consumeLong();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(String.format("%sL", result));
+      }
       return result;
     } else if (type == float.class || type == Float.class) {
       float result = data.consumeFloat();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(String.format("%sF", result));
+      }
       return result;
     } else if (type == double.class || type == Double.class) {
       double result = data.consumeDouble();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(Double.toString(result));
+      }
       return result;
     } else if (type == boolean.class || type == Boolean.class) {
       boolean result = data.consumeBoolean();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(Boolean.toString(result));
+      }
       return result;
     } else if (type == char.class || type == Character.class) {
       char result = data.consumeChar();
-      if (visitor != null)
+      if (visitor != null) {
         visitor.addCharLiteral(result);
+      }
       return result;
     }
     // Sometimes, but rarely return null for non-primitive and non-boxed types.
@@ -383,8 +391,9 @@ public class Meta {
     }
     if (type == String.class || type == CharSequence.class) {
       String result = data.consumeString(consumeArrayLength(data, 1));
-      if (visitor != null)
+      if (visitor != null) {
         visitor.addStringLiteral(result);
+      }
       return result;
     } else if (type.isArray()) {
       if (type == byte[].class) {
@@ -525,8 +534,9 @@ public class Meta {
       }
       return enumValue;
     } else if (type == Class.class) {
-      if (visitor != null)
+      if (visitor != null) {
         visitor.pushElement(String.format("%s.class", YourAverageJavaClass.class.getName()));
+      }
       return YourAverageJavaClass.class;
     } else if (type == Method.class) {
       if (visitor != null) {
@@ -716,7 +726,7 @@ public class Meta {
     Object[] result;
     try {
       result = Arrays.stream(executable.getGenericParameterTypes())
-                   .map((type) -> consume(data, type, visitor))
+                   .map(type -> consume(data, type, visitor))
                    .toArray();
       return result;
     } catch (AutofuzzConstructionException e) {

--- a/src/test/java/com/code_intelligence/jazzer/autofuzz/MetaTest.java
+++ b/src/test/java/com/code_intelligence/jazzer/autofuzz/MetaTest.java
@@ -195,6 +195,18 @@ public class MetaTest {
             (byte) 1, // do not return null
             0 /* first (and only) constructor*/));
   }
+
+  Class<?>[] returnsClassArray() {
+    throw new IllegalStateException(
+        "Should not be called, only exists to construct its generic return type");
+  }
+
+  @Test
+  public void testGetRawType() throws NoSuchMethodException {
+    Type classArrayType =
+        MetaTest.class.getDeclaredMethod("returnsClassArray").getGenericReturnType();
+    assertEquals(Class[].class, Meta.getRawType(classArrayType));
+  }
 }
 
 class OpinionatedClass {

